### DISCUSSION TITLE
Passphrase BIP39 entry

### DISCUSSION THREAD

**Discussion by hagagit** [selfcustody/krux#479] [Ideas]
I have got a Maix Cube. Works great, but the passphrase entry is a bit cumbersome. I have got a 7 word BIP39 passphrase. This has to be fully typed each time. Would it be possible that the passphrase entry works similar to the seed entry. That characters are preselected and you have to type max 4 chars per word?
I understand not everybody has a BIP39 passphrase, but could this be a seperate menue entry ( type BIP39 passphrase)

Regards and thanks for the great work.
Marcus

**Comment by odudex** [comment]
We could add this option.
Meanwhile, consider using tools to create a QR code of your passphrase, you can transcribe or export it as .bmp to SD card. Consider also a touchscreen device like Yahboom or WonderMV.

**Comment by jdlcdl** [comment]
My first thoughts are that this would reduce the strength of the passphrase the closer this gets to making it "easy" for data entry.
* if only the unique chars are needed, then entering whole words or partial words are exactly the same, while a single character change would normally be a few more bits of entropy.
* if shifted to lowercase, same thing.
* even if words are selected from a particular bip39 word-list (which severely limits the strength of the passphrase because it's effectively a smaller alphabet), 7 words is not "bip39", so I wonder where the requests for furthering special treatment of passphrase would end, because some users would complain that "an invalid bip39 mnemonic isn't very bip39-ish".

There have been discussions of krux supporting encrypted qrcodes... either generic bytes or text.  If there were an existing specification for doing this, or one that might be so simple that it would be easy for others to adopt, that seems like a better solution to simplify data entry, especially since it could be done after a wallet already exists and is funded.

**Comment by odudex** [comment]
Yes, ease of use often comes with tradeoffs in the security model or sovereignty. In @hagagit's case, I believe using a touchscreen device would significantly simplify things, eliminating the need to create a special entry mode and assess all its potential implications.

**Comment by tadeubas** [comment]
I think CC allows this type of passphrase using words.

https://coldcard.com/docs/passphrase/

**Comment by hagagit** [comment]
Thanks for the feedback. I really like the small formfactor and ease of use of the Cube. But I agree a touchscreen device is easier for the passphrase entry. But people use these small devices, and making the passphrase input difficult leads to shorter and weaker passphrases. 
A limit to words from the bip39 list is of course less secure than from a bigger list. But I believe 7 words from this list is enough entropy for a passphrase. And using these words allows to reuse the same procedure as used for the seed words. I can dice them the same way, and the input could be done in the same way as with the seed (type support). I do not think this support does reduce the security. Same as with the seeds.
I thing other signing devices also do it like that. Although I am not sure.
